### PR TITLE
WIP: Trying to add controller, view, capybara spec

### PR DESCRIPTION
**Pull request by joemsak**
Hey @solnic I am trying to learn about your ROM solution and using ROM-rails

I am still very much in over my head here, so please forgive me if I haven't understood something completely. I'm hoping to figure out how to build a rails app with your solution, hoping to contribute something meaningful back to your project.

So, I'd like if you could take a moment to check out my version of the code here, where my test is failing because it says there are no tuples in the relation after a seemingly successful call to create.

A couple notes about things that tripped me up along the way:
- I had to find out by luck of google search that I have to make a subclass of `ROM::Model::Form` to work with `form_for`
- I wasn't able to `redirect_to @user` I had to specifically `redirect_to user_path(@user)`
- If you don't at least provide a blank `validations { }` call in your `ROM::Model::Form` subclass, `rom-sql` blows up at you on `commands/create.rb:25`

I'll update this if I think of any more useful notes.

Would be awesome if you ran my spec and were able to figure out where my misunderstanding is and if I can help update your documentation or shed some light on issues that trip up someone completely new to your approach.

I really, really like what ROM seems to stand for and seems to me to want to help me achieve, so I hope to be helpful in both of us meeting our goals.

Thanks a lot for your work on this!

### CODE REVIEW

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -24,6 +24,7 @@ def conn
   config.include(MyAppHelper)
 
   config.before(:suite) do
+    `rake db:migrate`
```
Comment (joemsak) [inline spec/rails_helper.rb:27]: This is my stop-gap solution to provide the same functionality as `ActiveRecord::Migration.maintain_test_schema!` by the way

File: spec/features/users/create_users_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+RSpec.feature "Creating users" do
+  scenario "create a valid user" do
+    visit new_user_path
+
+    fill_in "Name", with: "Jane Doe"
+    fill_in "Email", with: "jane@doe.com"
+    click_button "Save"
+
+    expect(current_path).to eq(user_path(User.last))
```
Comment (joemsak) [inline spec/features/users/create_users_spec.rb:11]: I'm assuming `User.last` won't work here. That's fine... for now the test bombs at `click_button "Save"` anyway, when trying to redirect to the show action and get the user by id

**Comment by solnic** [comment]
rom-model will either go away or its internals will be replaced with dry-validation, and as far as our Form abstraction goes - this didn't turn out to be a good idea so we're looking into a better solution, like using Reform or something similar with ROM as the persistence backend.

For now I would like to suggest looking into using dry-validation for data validation and rom commands for persistence, standalone. If you want you can grab rom-repository from master (requires rom and rom-sql from master too) and check out its command support, where simple CUD ops boil down to `user_repo.create(name: "Joe")`.

What I'm trying to say is that we don't have a robust integration with Rails yet. rom-rails works OK but we're missing a good integration with its ActiveModel API (so that things can work with helpers etc.).

**Comment by joemsak** [comment]
Thank you for the response. I have installed the master versions of the gems you mentioned and I am at the drawing board once again trying to get it going :) Having a hard time learning how to create migrations at the moment but with some persistence (ha!) and sleeping on it I hope to get a better grip on things soon. I eagerly await the book!

**Pull request closed without merging**